packages/shared/src/components/cards/common/CardCover.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { ReactElement, useState } from 'react';
+import { ImageProps, ImageType } from '../../image/Image';
+import VideoImage, { VideoImageProps } from '../../image/VideoImage';
+import ConditionalWrapper from '../../ConditionalWrapper';
+import { CardImage } from '../Card';
+import { CardCoverShare } from './CardCoverShare';
+import { CommonCardCoverProps } from '../common';
+
+interface CardCoverProps extends CommonCardCoverProps {
+  imageProps: ImageProps;
+  videoProps?: Omit<VideoImageProps, 'imageProps'>;
+  isVideoType?: boolean;
+}
+
+export function CardCover({
+  imageProps,
+  videoProps,
+  justUpvoted,
```

Comment:
I see lots of props drilling to pull `justUpvoted` down to this component, maybe you can hook up into upvote mutatation with `useMutationSubscription` from the `CardCover` and keep the state there.

Reply:
Updated here https://github.com/dailydotdev/apps/pull/2705/commits/249608c50c4bd49604d5bbf4944c6b21018ee376